examples/n64lib/src/allocator.rs

```diff
@@ -0,0 +1,54 @@
+use crate::lock::interrupt_lock;
+use core::alloc::{GlobalAlloc, Layout};
+
+#[link(name = "c")]
+extern "C" {
+    fn malloc(size: usize) -> *mut u8;
+    fn free(ptr: *mut u8);
+}
+
+struct N64LibAlloc;
+
+unsafe impl GlobalAlloc for N64LibAlloc {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let _lock = interrupt_lock();
+
+        malloc(layout.size())
```

Comment:
This should probably assert that the requested layout alignment can actually be satisfied by `malloc`.

Reply:
Good catch. I think we can use `memalign(3)` instead of `malloc` here then.

From newlib: https://github.com/bminor/newlib/blob/6497fdfaf41d47e835fdefc78ecb0a934875d7cf/newlib/libc/stdlib/mallocr.c#L47